vanetza/asn1/fetch_ieee_asn1_files.cmake

```diff
@@ -0,0 +1,116 @@
+    "${DIRECTORY}/IEEE1609dot2dot1AcaRaInterface.asn")
+
+file(DOWNLOAD
+    https://forge.etsi.org/rep/ITS/asn1/ieee1609.2.1/-/raw/2022-published/Ieee1609Dot2Dot1Acpc.asn
+    "${DIRECTORY}/IEEE1609dot2dot1Acpc.asn")
+
+file(DOWNLOAD
+    https://forge.etsi.org/rep/ITS/asn1/ieee1609.2.1/-/raw/2022-published/Ieee1609Dot2Dot1CamRaInterface.asn
+    "${DIRECTORY}/IEEE1609dot2dot1CamRaInterface.asn")
+file(DOWNLOAD
+    https://forge.etsi.org/rep/ITS/asn1/ieee1609.2.1/-/raw/2022-published/Ieee1609Dot2Dot1CertManagement.asn
+    "${DIRECTORY}/IEEE1609dot2dot1CertManagement.asn")
+
+file(DOWNLOAD
+    https://forge.etsi.org/rep/ITS/asn1/ieee1609.2.1/-/raw/2022-published/Ieee1609Dot2Dot1EcaEeInterface.asn
+    "${DIRECTORY}/IEEE1609dot2dot1EcaEeInterface.asn")
+
+file(DOWNLOAD
+    https://forge.etsi.org/rep/ITS/asn1/ieee1609.2.1/-/raw/2022-published/Ieee1609Dot2Dot1EeMaInterface.asn
+    "${DIRECTORY}/IEEE1609dot2dot1EeMaInterface.asn")
+
+file(DOWNLOAD
+    https://forge.etsi.org/rep/ITS/asn1/ieee1609.2.1/-/raw/2022-published/Ieee1609Dot2Dot1EeRaInterface.asn
+    "${DIRECTORY}/IEEE1609dot2dot1EeRaInterface.asn")
+
+file(DOWNLOAD
+    https://forge.etsi.org/rep/ITS/asn1/ieee1609.2.1/-/raw/2022-published/Ieee1609Dot2Dot1LaMaInterface.asn
+    "${DIRECTORY}/IEEE1609dot2dot1LaMaInterface.asn")
+
+file(DOWNLOAD
+    https://forge.etsi.org/rep/ITS/asn1/ieee1609.2.1/-/raw/2022-published/Ieee1609Dot2Dot1LaRaInterface.asn
+    "${DIRECTORY}/IEEE1609dot2dot1LaRaInterface.asn")
+
+file(DOWNLOAD
+    https://forge.etsi.org/rep/ITS/asn1/ieee1609.2.1/-/raw/2022-published/Ieee1609Dot2Dot1MaRaInterface.asn
+    "${DIRECTORY}/IEEE1609dot2dot1MaRaInterface.asn")
+
+file(DOWNLOAD
+    https://forge.etsi.org/rep/ITS/asn1/ieee1609.2.1/-/raw/2022-published/Ieee1609Dot2Dot1Protocol.asn
+    "${DIRECTORY}/IEEE1609dot2dot1Protocol.asn")
+file(READ "${DIRECTORY}/IEEE1609dot2dot1Protocol.asn" _content)
+set(_valid_scoped_certificate_request "ScmsPdu-RaAcaCertRequest ::= ScmsPdu-Scoped {
+  AcaRaInterfacePdu (WITH COMPONENTS {
+    raAcaCertRequest
+  })
+}
+ScmsPdu-EeEcaCertRequest ::= ScmsPdu-Scoped {
+  EcaEeInterfacePdu (WITH COMPONENTS {
+    eeEcaCertRequest
+  })
+} 
+ScmsPdu-EeRaCertRequest ::= ScmsPdu-Scoped {
+  EeRaInterfacePdu (WITH COMPONENTS {
+    eeRaCertRequest
+  })
+}
+ScmsPdu-EeRaSuccessorEnrollmentCertRequest ::= ScmsPdu-Scoped {
+  EeRaInterfacePdu (WITH COMPONENTS {
+    eeRaSuccessorEnrollmentCertRequest
+  })
+}
+
+
+/**
+* @brief This structure defines the all certificate request structures as a
+* scoped version of the ScmsPdu.
+*/
+ScopedCertificateRequest ::= ScmsPdu (
+ScmsPdu-RaAcaCertRequest | ScmsPdu-EeEcaCertRequest | 
+ScmsPdu-EeRaCertRequest  | ScmsPdu-EeRaSuccessorEnrollmentCertRequest
+)
+/**
+ * @brief This structure defines the format of a signed certificate request.")
+string(REGEX REPLACE "ScopedCertificateRequest ::=.+signed certificate request." "${_valid_scoped_certificate_request}" _content "${_content}")
```

Comment:
It becomes quite hard to see the actual patch this way. I prefer to have a patch file then. Is there a reason why a patch file is inappropriate here?

Reply:
No, the complexity only started to explode after beginning that script already so i didn't change it to a patch, but good idea!